scenarios/bigquery_test.py

```diff
@@ -17,55 +17,54 @@
-            with open(self.out_filename) as out_file:
-                contents = out_file.read()
-                if "error" in contents.split("\n")[0].lower():
-                    raise ValueError("query should have thrown an error because the"
-                                     " bq command failed.")
 
     def test_jq(self):
         """Test that the do_jq function can execute a jq filter properly."""
         # [filter, data, expected output]
-        tests = [[".", '{ "field": "value" }', '{"field":"value"}'],
-                 [".field", '{ "field": "value" }', '"value"']]
+        tests = [['.', '{ "field": "value" }', '{"field":"value"}'],
+                 ['.field', '{ "field": "value" }', '"value"']]
         for test in tests:
-            with open(self.data_filename, "w") as data_file:
+            with open(self.data_filename, 'w') as data_file:
                 data_file.write(test[1])
-            bigquery.do_jq(test[0], self.data_filename, self.out_filename)
+            bigquery.do_jq(test[0], self.data_filename, self.out_filename, jq_bin=ARGS.jq)
             with open(self.out_filename) as out_file:
-                actual = out_file.read().replace(" ", "").replace("\n", "")
-                self.assertEqual(actual, test[2], msg="expected jq '{}' on data: {} to output {}"
-                                 " but got {}".format(test[0], test[1], test[2], actual))
+                actual = out_file.read().replace(' ', '').replace('\n', '')
+                self.assertEqual(actual, test[2], msg='expected jq "{}" on data: {} to output {}'
+                                 ' but got {}'.format(test[0], test[1], test[2], actual))
 
     def test_validate_metric_name(self):
         """Test the the validate_metric_name function rejects invalid metric names."""
-        tests = ["invalid#metric", "invalid/metric", "in\\valid", "invalid?yes", "*invalid",
-                 "[metric]", "metric\n", "met\ric"]
+        tests = ['invalid#metric', 'invalid/metric', 'in\\valid', 'invalid?yes', '*invalid',
+                 '[metric]', 'metric\n', 'met\ric']
         for test in tests:
             self.assertRaises(ValueError, bigquery.validate_metric_name, test)
 
     def setUp(self):
-        self.tmpdir = tempfile.mkdtemp(prefix="bigquery_test_")
-        os.chdir(self.tmpdir)
+        self.assertTrue(ARGS.jq)
+        self.tmpdir = tempfile.mkdtemp(prefix='bigquery_test_')
+        self.out_filename = os.path.join(self.tmpdir, 'out.json')
+        self.data_filename = os.path.join(self.tmpdir, 'data.json')
 
     def tearDown(self):
         shutil.rmtree(self.tmpdir)
 
-if __name__ == "__main__":
-    unittest.main()
+if __name__ == '__main__':
+    PARSER = argparse.ArgumentParser()
+    PARSER.add_argument('--jq',
+                        required=True,
+                        type=str,
+                        help='The path to the "jq" command.')
+    ARGS = PARSER.parse_args()
+
+    RESULTS = unittest.TextTestRunner().run(
```

Comment:
unittest.main() uses the command line flags and throws an error when it receives unknown flags so I was just running a TestRunner instead to avoid complaints about the '--jq' flag that this test script uses to get the install location of jq from bazel.  I made a change that removes the '--jq' flag from sys.argv after the known args have been parsed and before unittest.main is called.  Calls like `./bigquery_test.py --jq jq TestBigquery.test_validate_metric_name` now work properly.